alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs

```diff
@@ -138,51 +139,34 @@ transAddr :: Addr crypto -> Maybe P.Address
 transAddr (Addr _net object stake) = Just (P.Address (transCred object) (transStakeReference stake))
 transAddr (AddrBootstrap _bootaddr) = Nothing
 
--- ===============================
--- Translate ValidityIntervals
--- TODO remove this function when we are using transVITime instead
-transVI ::
+slotToPOSIXTime ::
   EpochInfo Identity ->
   SystemStart ->
-  ValidityInterval ->
-  P.SlotRange
-transVI _ _ (ValidityInterval SNothing SNothing) =
-  P.Interval
-    (P.LowerBound P.NegInf True)
-    (P.UpperBound P.PosInf False)
-transVI _ _ (ValidityInterval (SJust (SlotNo i)) SNothing) =
-  P.Interval
-    (P.LowerBound (P.Finite (fromIntegral i)) True)
-    (P.UpperBound P.PosInf False)
-transVI _ _ (ValidityInterval SNothing (SJust (SlotNo i))) =
-  P.Interval
-    (P.LowerBound P.NegInf True)
-    (P.UpperBound (P.Finite (fromIntegral i)) False)
-transVI _ _ (ValidityInterval (SJust (SlotNo i)) (SJust (SlotNo j))) =
-  P.Interval
-    (P.LowerBound (P.Finite (fromIntegral i)) True)
-    (P.UpperBound (P.Finite (fromIntegral j)) False)
+  SlotNo ->
+  P.Extended P.POSIXTime
+slotToPOSIXTime ei sysS s =
+  P.Finite . P.POSIXTime . floor . toRational . utcTimeToPOSIXSeconds . runIdentity $
```

Comment:
I would love to use something besides `floor . toRational` to convert `NominalDiffTime` to an integer, but I could not find anything better. Any ideas?